lms/djangoapps/courseware/grades.py

```diff
@@ -126,9 +127,22 @@ def grade(student, request, course, keep_raw_scores=False):
     """
     Wraps "_grade" with the manual_transaction context manager just in case
     there are unanticipated errors.
+    Send a signal to update the minimum grade requirement status.
     """
     with manual_transaction():
-        return _grade(student, request, course, keep_raw_scores)
+        grade_summary = _grade(student, request, course, keep_raw_scores)
+        responses = MIN_GRADE_REQUIREMENT_STATUS.send_robust(
+            sender=None,
+            username=request.user.username,
+            grade_summary=grade_summary,
+            course_key=course.id,
+            deadline=course.end
+        )
+
+        for receiver, response in responses:
+            log.info('Sent signal to receiver: %s. Response: %s', receiver, response)
```

Comment:
It's probably worth saying in the log message that this is a signal fired when a student's grade is calculated.